src/include/sof/audio/volume.h

```diff
@@ -34,15 +34,30 @@
 struct comp_buffer;
 struct sof_ipc_ctrl_value_chan;
 
-#define CONFIG_GENERIC
-
 #if defined(__XCC__)
 #include <xtensa/config/core-isa.h>
 
-#if XCHAL_HAVE_HIFI3
-#undef CONFIG_GENERIC
+#if XCHAL_HAVE_HIFI4  /* HiFi 4 */
```

Comment:
Btw, XCHAL is xtos specific, we need a high level CONFIG_ARCH_XTENSA_HIFI4 etc and use them instead. @singalsu fyi.

Reply:
XCHAL_HAVE_xxx is provided by Xtensa toolchain (processror config), we have a copy in src/platform/core_code_name/include/arch just for gcc complication.  it is toolchain specific indeed.